cachet-tools/update_cachet.py

```diff
@@ -0,0 +1,114 @@
+"""Cachet Updater
+This script updates status in Cachet by first querying Cachet for services,
+checks the status, each service discovered, on Insights, and then updates Cachet with
+latest status.
+"""
+
+import json
+import logging
+import os
+import requests
+import time
+
+from services import Services
+
+# target cachet listing insights services 
+CACHET_HOSTNAME = os.environ.get("CACHET_HOSTNAME")
+CACHET_TOKEN    = os.getenv("CACHET_TOKEN")
+CACHET_URL      = f"https://{CACHET_HOSTNAME}/api/v1/components"
+HEADERS         = { 'X-Cachet-Token': os.environ.get("CACHET_TOKEN") }
+
+# insights server hosting live services status
+INSIGHTS_USERNAME = os.getenv("INSIGHTS_USERNAME")
+INSIGHTS_PASSWORD = os.getenv("INSIGHTS_PASSWORD")
+INSIGHTS_SERVER   = os.getenv("INSIGHTS_SERVER")
+API_URL           = f"https://{INSIGHTS_SERVER}/api"
+
+# format the logger message
+logging.basicConfig(level='INFO', format='%(asctime)s %(message)s', datefmt='%m/%d/%Y %H:%M:%S %p: ')
+logger = logging.getLogger(__name__)
+
+# session used by every request, only URLs and args will change
+session  = requests.Session()
+services = Services()
+
+def update_all_services():
+    logger.info("Starting services status check ...")
+
+    # session = requests.Session()
+    session.headers.update(HEADERS)
+
+    try:
+        response = session.get(CACHET_URL + "/groups", verify=False)
+        groups   = response.json()['data']
+        logger.info("Number of groups found: " + str(len(groups)))
+
+        for group in groups:  # for somereason not using groups returns 20 components but should return 23
+            components = group['enabled_components']
+            logger.info("Starting to check each of " + str(len(components)) + " components in group \"" + group['name'] + "\"")
+            for component in components:
+                # assuming the component in the parent group is accessible because it showed up.
+                svc_status = get_service_status(component['name'])
+                update_svc = set_svc_status(component['id'], svc_status)
+                if update_svc:
+                    logger.info("Successfully updated status of \"" + component['name'] + "\" service")
+                else:
+                    logger.error("Failed to updated status of \"" + component['name'] + "\" service")
+                time.sleep(2) 
+            # end of for components loop
+        # end of for groups loop
+    except Exception as ex:
+        logger.exception(ex)
+    # end of try/except block
+
+    logger.info("Done updating all services!!!")
+# end of def update_all_services
+
+def get_service_status(name):
```

Comment:
Can we rename this function to something like `determine_status` or something?  The current name makes me think you are getting the status from cachet.